### PR TITLE
User/sid/g#23

### PR DESCRIPTION
**Pull request by SidhaantAnand**
Removed new similarity
Fixed Update similarity
Added User surprise db update

### CODE REVIEW

File: back/src/ML/app.js [hunk 1]
```diff
@@ -61,6 +61,21 @@ function post(req,res,constants,request,client) {
 		}		
 	}
 
+	else if(req.params.type == 'surprise') {
```
Comment (sohinim98) [inline back/src/ML/app.js]: use '===' everywhere

File: back/src/ML/app.js [hunk 2]
```diff
@@ -61,6 +61,21 @@ function post(req,res,constants,request,client) {
...
+		if(req.params.what == 'update') {
+			if(!req && !req.body && !req.body.data) {
```
Comment (sohinim98) [inline back/src/ML/app.js:66]: use !req?.body?.data if you can.

File: back/src/ML/routes/update_similarity.js [hunk 3]
```diff
@@ -7,13 +7,23 @@ module.exports = function(req,res,constants,client,helpers) {
 			const id2 = curr.user2;
 			const similarity = curr.cosine_similarity;
 
-			var friends_id1  = users[db_ids.indexOf(id1)].friends;
-			var similarity_id1 = users[db_ids.indexOf(id1)].similarity;
-			var friends_id2  = users[db_ids.indexOf(id2)].friends;
-			var similarity_id2 = users[db_ids.indexOf(id2)].similarity;
+			var friends_id1 = [];
```
Comment (sohinim98) [inline back/src/ML/routes/update_similarity.js]: change all vars to let or const.

File: back/src/ML/routes/update_similarity.js [hunk 4]
```diff
@@ -7,13 +7,23 @@ module.exports = function(req,res,constants,client,helpers) {
...
+			var similarity_id1 = [];
+			var friends_id2 = [];
+			var similarity_id2 = [];
 
-			var index_id1 = 0;
-			var index_id2 = 0;
+			if(db_ids.indexOf(id1) != -1) {
```
Comment (sohinim98) [inline back/src/ML/routes/update_similarity.js]: change to !== everywhere.

File: back/src/ML/routes/update_similarity.js [hunk 5]
```diff
@@ -29,19 +39,32 @@ module.exports = function(req,res,constants,client,helpers) {
 				}
 			}
 
-			friends_id1[index_id1] = id2;
-			similarity_id1[index_id1] = similarity;
+			if(index_id1 == -1) {
```
Comment (sohinim98) [inline back/src/ML/routes/update_similarity.js]: Change to ===.

**Review by sohinim98 — CHANGES_REQUESTED** [review]

File: back/src/ML/app.js (hunk 2 above)
Comment (SidhaantAnand) [reply back/src/ML/app.js:66]: thats ts
apart from that I have made the changes

**Review by sohinim98 — APPROVED** [review]

**Comment by sohinim98** [comment]
lgtm

**Pull request merged**